src/ctia/schemas/search_agg.clj

```diff
@@ -65,3 +65,10 @@
   (st/open-schema
    {:agg-type AggType
     :aggregate-on s/Str}))
+
+(s/defschema AggResult
+  {:data {s/Keyword s/Any}
+   :type AggType
+   :filters (st/open-schema
+             {:from s/Inst
+              :to s/Inst})})
```

Comment:
As I said in a previous comment, you might be able to take advantage of generating
a few more precise schemas for each kind of aggregation.